services/motion/builtin/move_request.go

```diff
@@ -154,8 +166,20 @@ func (mr *moveRequest) execute(ctx context.Context, plan motionplan.Plan) (state
 		return state.ExecuteResponse{}, err
 	}
 
-	// the plan has been fully executed so check to see if where we are at is close enough to the goal.
-	return mr.deviatedFromPlan(ctx, plan)
+	// the plan has been fully executed, so check to see if where we are at is close enough to the goal
+	executionState, err := mr.kinematicBase.ExecutionState(ctx)
+	if err != nil {
+		return state.ExecuteResponse{}, err
+	}
+	currentPosition, ok := executionState.CurrentPoses()[mr.kinematicBase.LocalizationFrame().Name()]
+	if !ok {
+		return state.ExecuteResponse{}, errors.New("exeuctionState.CurrentPoses() does not contain an entry for the LocalizationFrame")
+	}
+	resp = mr.alreadyAtGoalCheck(currentPosition.Pose())
+	if resp == nil {
```

Comment:
Here we are returning an ExecutionResponse and immediately discarding it in favor of constructing a new one.  For that reason, we should change the function to only return a bool

Reply:
Another option would be moving this logic into the function itself, and writing `return resp, resp != nil`

Reply:
> we should change the function to only return a bool

This is a good idea. It will simplify the function.